### ISSUE TITLE
Generator.prototype is not defined

### ISSUE THREAD

**Issue by arai-a** [tc39/ecma262#3230]
The "Generator Objects" section first mentions "A Generator is an instance of a generator function ...".

https://tc39.es/ecma262/#sec-generator-objects
> A Generator is an instance of a generator function and conforms to both the Iterator and Iterable interfaces.

Then "Properties of the Generator Prototype Object" section defines "Generator prototype object" as `%GeneratorFunction.prototype.prototype%`

https://tc39.es/ecma262/#sec-properties-of-generator-prototype
> The Generator prototype object:
>   * is %GeneratorFunction.prototype.prototype%.

The "Generator prototype object" name is used by `GeneratorFunction.prototype.prototype`'s definition, which makes sense:

https://tc39.es/ecma262/#sec-generatorfunction.prototype.prototype
> The initial value of GeneratorFunction.prototype.prototype is the [Generator prototype object](https://tc39.es/ecma262/#sec-properties-of-generator-prototype).

But remaining subsections of "Generator Objects" define properties of `Generator.prototype`, that sounds like the same thing as `%GeneratorFunction.prototype.prototype%`, but it's not defined explicitly:

https://tc39.es/ecma262/#sec-generator.prototype.constructor
> 27.5.1.1 Generator.prototype.constructor

https://tc39.es/ecma262/#sec-generator.prototype.next
> 27.5.1.2 Generator.prototype.next ( value )

It also makes the first paragraph "A Generator is an instance of ..." confusing, that it sounds like `Generator` in `Generator.prototype.constructor` is "A Generator" instance (which is not true, because the instance doesn't have `prototype` property).

There would be 2 options to solve this:
  * (A) Replace all occurrences of `Generator.prototype` with `%GeneratorFunction.prototype.prototype%` (while keeping the ids of those sections, to avoid breaking links)
  * (B) Add a section or a paragraph that defines that `Generator.prototype` is a shorthand notation for `%GeneratorFunction.prototype.prototype%` (also clarifies `Generator` there isn't "A Generator" above)

Same for `AsyncGenerator.prototype` in https://tc39.es/ecma262/#sec-asyncgenerator-objects

**Comment by arai-a** [comment]
I was assuming that "Generator" means the `g()` value for `function *g() {}`. Isn't it correct?  Does it mean the `g` value ?

**Comment by ljharb** [comment]
> A Generator is an instance of a generator function and conforms to both the Iterator and Iterable interfaces.

I read that as the return value of `g()`, there, yes.

**Comment by arai-a** [comment]
In that case, `Generator.prototype` is `undefined`, given `g().prototype` is `undefined`.

**Comment by ljharb** [comment]
ahh ok, on a reread "instance of a generator function" actually means `g` itself, yes. I see how that's confusing as I was myself confused.

It's still defined, though :-)

**Comment by ljharb** [comment]
I agree with you that some terminology needs clarifying in this section.
